batcher/sweep_batcher.go

```diff
@@ -0,0 +1,286 @@
+package sweepbatcher
+
+import (
+	"context"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/lndclient"
+	"github.com/lightninglabs/loop/loopdb"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwallet/chainfee"
+)
+
+const (
+	// DefaultMaxTimeoutDistance is the default maximum timeout distance
+	// of sweeps that can appear in the same batch.
+	DefaultMaxTimeoutDistance = 288
+)
+
+type BatcherStore interface {
+	// FetchUnconfirmedSweepBatches fetches all the batches from the database that are
+	// not in a confirmed state.
+	FetchUnconfirmedSweepBatches(ctx context.Context) ([]*loopdb.Batch, error)
+
+	// UpsertSweepBatch inserts a batch into the database, or updates an existing batch
+	// if it already exists.
+	UpsertSweepBatch(ctx context.Context, batch *loopdb.Batch) error
+
+	// ConfirmBatch confirms a batch by setting its state to confirmed.
+	ConfirmBatch(ctx context.Context, id []byte) error
+}
+
+type MuSig2SignSweep func(ctx context.Context,
```

Comment:
nit: could use a functional options pattern or a config here to help readability.

Reply:
I don't have a strong opinion here
I think it the config approach fits well with `NewX` methods, but maybe not so well with single specific methods like this?
cc @bhandras @sputn1ck